crates/vhost-device-gpio/src/mock_gpio.rs

```diff
@@ -0,0 +1,160 @@
+// Mock GPIO backend device for testing
+//
+// Copyright 2023 Linaro Ltd. All Rights Reserved.
+//          Viresh Kumar <viresh.kumar@linaro.org>
+//
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use crate::gpio::{Error, GpioDevice, GpioState, Result};
```

Comment:
Can you also organize this in the 3 sections mentioned here: https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=#StdExternalCrate\%3A

We are following this, maybe we should enforce in the CI